### PR TITLE
feat(git): per-machine identity + lazygit + gh + XDG_CONFIG_HOME

### PR DESCRIPTION
**Pull request by kyriekevin**
## Summary / 概述

Phase 6a — minimal global git config (identity switched per-machine via `is_work`, not per-directory), plus the companion tools users actually drive from the CLI/TUI: `lazygit` (Catppuccin Mocha + delta pager) and `gh` (PR-review aliases). Sets `XDG_CONFIG_HOME=$HOME/.config` in `dot_zshenv` so lazygit reads the same path on macOS as on Linux.

## Change type / 变更类型

- [x] `feat` — new package / functionality · 新包或新功能
- [ ] `fix` — bug fix · bug 修复
- [ ] `chore` — infrastructure, hooks, dependency bumps · 基础设施 / hooks / 依赖升级
- [ ] `docs` — documentation only · 纯文档
- [ ] `refactor` — no behavior change · 纯重构（无行为变化）

## Checklist / 检查清单

- [x] Commit messages follow Conventional Commits · Commit 信息符合 Conventional Commits
- [x] `pre-commit run --all-files` passes locally · 本地跑过 `pre-commit run --all-files` 且全绿
- [x] `chezmoi diff` reviewed; no surprising deletions or permission changes · 已 review `chezmoi diff`，没有意外的删除或权限变化
- [ ] New package → entry added to the `Layout` section of `README.md` · 新包已加入 `README.md` 的 `Layout` 小节
  - README `Layout` uses abstract `dot_*` / `*.tmpl` placeholders, not per-package lines — no drift. · README Layout 用的是抽象 `dot_*` / `*.tmpl`，不是逐包列，没 drift。
- [x] New CLI tool → added to `Brewfile` · 新 CLI 工具已加入 `Brewfile`
  - `git-delta`, `lazygit`, `gh` already shipped in Phase 2 Brewfile. · `git-delta`、`lazygit`、`gh` Phase 2 Brewfile 已经带了。
- [x] Smoke-tested end-to-end on at least one Mac · 在至少一台 Mac 上做过端到端冒烟
  - `tests/git.sh` 44/44 green; `lazygit --print-config-dir` = `~/.config/lazygit`; `gh alias list` lists all 8 aliases; new shell reads `XDG_CONFIG_HOME`.

## Notes for reviewer / 给 reviewer 的说明

**Identity strategy simplification.** Original plan had per-directory `includeIf` splitting (`config.personal` / `config.work`). Dropped in favor of `{{ if .is_work }}` in `dot_gitconfig.tmpl` — user's actual habit is "whichever Mac I'm on, commit as that Mac"; projects mix freely, no directory-based identity needed. Saved memory: `project_git_identity.md`.

**`XDG_CONFIG_HOME` is a global env-var change in `dot_zshenv`.** Blast radius audit: every user-facing tool we track (starship, ghostty, yazi, fastfetch, zsh, karabiner) already stores config under `~/.config/`, so their paths don't shift. The visible effect is that lazygit — which on macOS defaults to `~/Library/Application Support/lazygit/` — now reads `~/.config/lazygit/`. Any third-party tool that was previously using a macOS-native config path and becomes XDG-aware will also move; user has no such tools currently tracked.

**lazygit schema fix (commit 3).** lazygit ≥ 0.40 auto-migrated `git.paging` → `git.pagers` on first launch (user flagged the "Moved git.paging object to git.pagers array" message). Source was then out of sync → next `chezmoi apply` would write old schema back → migration loop. `b3e7177` syncs source to the new schema so it closes.

**`hosts.yml` is pre-emptively ignored in `.chezmoiignore`**. It holds per-machine auth state (oauth_token itself is in macOS Keychain on recent gh, but the file references the keyring slot). Never tracked; each machine does `gh auth login` once.

**Phase 8/9 dropped from plan.** Pre-commit base (whitespace / gitleaks / conventional-commits) was done in Phase 0; `.github/` templates done in batch post-processing; remaining Phase-8 ideas (`ruff`, `validate-pyproject`) have no Python in this repo. Phase 9 cross-machine smoke is user-owned (no memory value). Plan file updated.

**Phase 6b (Claude config) lands in a follow-up PR**, not bundled here, per user request to split.

---

_Generated with Claude Code._

### CODE REVIEW

File: tests/git.sh [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+FAIL=0
+ok()  { printf "  \033[32m✓\033[0m %s\n" "$1"; PASS=$((PASS+1)); }
+bad() { printf "  \033[31m✗\033[0m %s\n" "$1"; [[ -n ${2:-} ]] && printf "    %s\n" "$2"; FAIL=$((FAIL+1)); }
+
+check() {
+    local name=$1 cmd=$2 out
+    if out=$(eval "$cmd" 2>&1); then ok "$name"; else bad "$name" "$out"; fi
+}
+
+SRC="/Users/bytedance/.dotfiles/dot_gitconfig.tmpl"
```
[the remark this answers is not in the document]
Comment (kyriekevin) [reply tests/git.sh]: Accepted in 00be918 — applied to all 5 hardcoded-path sites in this script, not just line 20. `REPO_ROOT="$(cd "$(dirname "$0")/.." && pwd)"` sits at the top, and every absolute reference uses `$REPO_ROOT/...` instead.

Verified runs clean from a different CWD:

```
$ cd /tmp && bash /Users/bytedance/.dotfiles/tests/git.sh | tail -1
  42 passed, 0 failed
```

Caveat on the "portability" framing: the repo is single-user macOS-only by explicit design (Brewfile, Karabiner, ghostty cask, the `is_work` prompt about "the work **Mac**"), so the practical win isn't "another user could run it" — it's "the script decouples from its install path" for future repo-location changes. Also now consistent with how every other phase test handles paths (they read targets from `$HOME`; none had the source-path bug this one had, because `tests/git.sh` is the only phase test that reads *source* files — to verify both template identity branches exist).

---

_Generated with Claude Code._

File: tests/git.sh [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+check "work name (zyz) in template"             "grep -q 'name  = zyz' $SRC"
+check "work email (bytedance) in template"      "grep -q 'email = zhongyuzhe@bytedance.com' $SRC"
+check "personal name (Kyrie) in template"       "grep -q 'name  = Kyrie' $SRC"
+check "personal email (qq) in template"         "grep -q 'email = yuzhezhong0117@qq.com' $SRC"
+
+echo
+echo "── lazygit config ───────────────────────────────────"
+# XDG_CONFIG_HOME must be set in zshenv, else lazygit reads
+# ~/Library/Application Support/lazygit/ on macOS and our config is dead.
+check "XDG_CONFIG_HOME set in dot_zshenv"       "grep -q 'export XDG_CONFIG_HOME=\"\$HOME/.config\"' /Users/bytedance/.dotfiles/dot_zshenv"
```
[the remark this answers is not in the document]
Comment (kyriekevin) [reply tests/git.sh]: Accepted — same fix as line 20. All 5 hardcoded `/Users/bytedance/.dotfiles` sites swapped to `$REPO_ROOT` in a single refactor in 00be918. See [full reply + verification on line 20](https://github.com/kyriekevin/dotfiles/pull/14#discussion_r3106331785).

---

_Generated with Claude Code._

File: tests/git.sh [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+echo "── gh config ────────────────────────────────────────"
+check "gh config.yml parses (yaml)"             "python3 -c 'import yaml,sys; yaml.safe_load(open(\"$GHCFG\"))'"
+check "gh git_protocol = https"                 "grep -q 'git_protocol: https' $GHCFG"
+check "gh alias 'co' (pr checkout)"             "grep -q 'co: pr checkout' $GHCFG"
+check "gh alias 'pv' (pr view)"                 "grep -q 'pv: pr view' $GHCFG"
+check "gh alias 'pc' (pr checks)"               "grep -q 'pc: pr checks' $GHCFG"
+# Cross-check: gh actually parses the aliases (not just YAML-valid but
+# gh-valid). `gh alias list` exits non-zero if the file is corrupt.
+check "gh alias list succeeds"                  "gh alias list >/dev/null"
+check "hosts.yml ignored in chezmoi source"     "grep -q 'dot_config/gh/hosts.yml' /Users/bytedance/.dotfiles/.chezmoiignore"
```
[the remark this answers is not in the document]
Comment (kyriekevin) [reply tests/git.sh]: Accepted — same fix as line 20. All 5 hardcoded `/Users/bytedance/.dotfiles` sites swapped to `$REPO_ROOT` in a single refactor in 00be918. See [full reply + verification on line 20](https://github.com/kyriekevin/dotfiles/pull/14#discussion_r3106331785).

---

_Generated with Claude Code._

File: tests/git.sh [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+echo
+echo "── Smoke ────────────────────────────────────────────"
+# Active identity matches the active machine's is_work — we're always on
+# one or the other, so whichever renders must be self-consistent.
+check "active identity matches template"        "git config --global user.email | grep -qE '^(zhongyuzhe@bytedance\.com|yuzhezhong0117@qq\.com)$'"
+check "git lg alias renders (cat pager)"        "git -C /Users/bytedance/.dotfiles -c core.pager=cat lg -1 >/dev/null"
```
[the remark this answers is not in the document]
Comment (kyriekevin) [reply tests/git.sh]: Accepted — same fix as line 20. All 5 hardcoded `/Users/bytedance/.dotfiles` sites swapped to `$REPO_ROOT` in a single refactor in 00be918. See [full reply + verification on line 20](https://github.com/kyriekevin/dotfiles/pull/14#discussion_r3106331785).

---

_Generated with Claude Code._

File: tests/git.sh [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+check "work email (bytedance) in template"      "grep -q 'email = zhongyuzhe@bytedance.com' $SRC"
+check "personal name (Kyrie) in template"       "grep -q 'name  = Kyrie' $SRC"
+check "personal email (qq) in template"         "grep -q 'email = yuzhezhong0117@qq.com' $SRC"
+
+echo
+echo "── lazygit config ───────────────────────────────────"
+# XDG_CONFIG_HOME must be set in zshenv, else lazygit reads
+# ~/Library/Application Support/lazygit/ on macOS and our config is dead.
+check "XDG_CONFIG_HOME set in dot_zshenv"       "grep -q 'export XDG_CONFIG_HOME=\"\$HOME/.config\"' /Users/bytedance/.dotfiles/dot_zshenv"
+check "lazygit config.yml parses (yaml)"        "python3 -c 'import yaml,sys; yaml.safe_load(open(\"$LG\"))'"
```
[the remark this answers is not in the document]
Comment (kyriekevin) [reply tests/git.sh]: Accepted in 00be918 — dropped both `python3 -c 'import yaml; yaml.safe_load(...)'` checks (lazygit and gh).

Your PyYAML-not-stdlib observation is correct, and you're also right that `gh alias list` already exercises gh's own schema structurally. Extending that logic: both config files are tracked in the repo tree, so pre-commit's `check-yaml` hook parses them at commit time — the runtime re-parse was pure redundancy. Now the test trusts pre-commit for syntax and `gh alias list` + field-specific greps for semantics:

```
$ pre-commit run check-yaml --all-files
check yaml...............................................................Passed
$ bash tests/git.sh | tail -1
  42 passed, 0 failed
```

Tests went 44 → 42; remaining coverage unchanged.

---

_Generated with Claude Code._

File: dot_gitconfig.tmpl [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+[delta]
+	navigate     = true
+	side-by-side = true
+	line-numbers = true
+
+[commit]
+	verbose = true
+
+[credential]
+	helper = osxkeychain
```
[the remark this answers is not in the document]
Comment (kyriekevin) [reply dot_gitconfig.tmpl:43]: Declining — out of scope. This repo is macOS-only by explicit design, not just happen-of-fact:

1. **`Brewfile` is macOS-specific.** `brew bundle` is Phase 2's single bootstrap path; there's no apt/dnf/pacman branch.
2. **Cask dependencies** — `ghostty`, `karabiner-elements` — only exist on macOS. Half the phases (4-5) are meaningless elsewhere.
3. **The `is_work` prompt literally says "Is this the work Mac?"** — not "work machine." The two-identity design in this PR maps to **"work Mac" ↔ "personal Mac"**, confirmed in `docs/git.md`'s identity table.
4. **README mentions Linux only to explain the XDG choice** — "Linux-style `~/.config/` everywhere" is about the config-path convention, not a claim of Linux support. There's no Linux bootstrap, no apt Brewfile equivalent, no cross-OS CI.

If cross-OS support ever gets added (it won't — deliberate 2-Mac scope), the right shape is a future `feat/linux` PR that wraps many things in `{{ if eq .chezmoi.os "darwin" }}`, not just `credential.helper`. Adding one conditional here would be aspirational — the rest of the file (and repo) would still break elsewhere, so the conditional gives a false sense of portability.

`credential.helper = osxkeychain` works correctly where this config is actually deployed. Verified:

```
$ git config --global credential.helper
osxkeychain
$ bash tests/git.sh | grep credential
  ✓ credential.helper = osxkeychain
```

---

_Generated with Claude Code._

[the remark this answers is not in the document]
Comment (kyriekevin) [reply dot_gitconfig.tmpl:43]: Same decision as round 1 — declining. No new evidence this round: the variant just swaps a block conditional for an inline ternary, both presuppose cross-OS support this repo explicitly doesn't target.

Re-stating the R1 scope argument: `Brewfile` + casks (`ghostty`, `karabiner-elements`) + the whole `feat/karabiner` phase + the `is_work` prompt wording ("Is this the work **Mac**?") all make this repo macOS-only by design, not accident. The XDG mention in `dot_zshenv`'s comment is about config-path convention, not a portability claim — reading it as one is pattern-matching from generic dotfiles.

See the [full 4-point argument + verification](https://github.com/kyriekevin/dotfiles/pull/14#discussion_r3106331784).

---

_Generated with Claude Code._

File: tests/git.sh [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+check "push.autoSetupRemote = true"             "test \"$(git config --global push.autoSetupRemote)\" = true"
+check "fetch.prune = true"                      "test \"$(git config --global fetch.prune)\" = true"
+check "rebase.autoStash = true"                 "test \"$(git config --global rebase.autoStash)\" = true"
+check "merge.conflictstyle = zdiff3"            "test \"$(git config --global merge.conflictstyle)\" = zdiff3"
+check "interactive.diffFilter uses delta"       "git config --global interactive.diffFilter | grep -q '^delta'"
+check "delta.navigate = true"                   "test \"$(git config --global delta.navigate)\" = true"
+check "delta.side-by-side = true"               "test \"$(git config --global delta.side-by-side)\" = true"
+check "delta.line-numbers = true"               "test \"$(git config --global delta.line-numbers)\" = true"
+check "commit.verbose = true"                   "test \"$(git config --global commit.verbose)\" = true"
+check "credential.helper = osxkeychain"         "test \"$(git config --global credential.helper)\" = osxkeychain"
```
[the remark this answers is not in the document]
Comment (kyriekevin) [reply tests/git.sh:57]: Declining — derivative of the cross-OS decline on line 43. As long as the config stays macOS-only (see [scope argument](https://github.com/kyriekevin/dotfiles/pull/14#discussion_r3106331784)), loosening this assertion from `= osxkeychain` to `-n "…"` trades a precise regression guard for a fuzzy one: it would pass for **any** non-empty helper value, including misconfigs like `manager-core` / typos / stale values that the exact-match check would flag immediately.

The test's job is "did chezmoi render what we pinned in `dot_gitconfig.tmpl`?" — aligning strictness between template and test is the whole point. If this repo ever goes cross-OS (it won't), the template would get an `{{ if eq .chezmoi.os "darwin" }}` conditional AND the test would get a matching OS switch — not a loosened test against a still-rigid template.

---

_Generated with Claude Code._
